pkg/types/validation/installconfig.go

```diff
@@ -595,6 +597,61 @@ func validateNetworkingClusterNetworkMTU(c *types.InstallConfig, fldPath *field.
 
 	return allErrs
 }
+func validateOVNKubernetesConfig(n *types.Networking, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if n.OVNKubernetesConfig == nil {
+		return nil
+	}
+
+	if !strings.EqualFold(n.NetworkType, string(operv1.NetworkTypeOVNKubernetes)) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networkType"), n.NetworkType, "ovnKubernetesConfig may only be specified with the OVNKubernetes networkType"))
+	}
+
+	allErrs = append(allErrs, validateV4InternalSubnet(n, fldPath.Child("ovnKubernetesConfig").Child("v4InternalSubnet"))...)
+	return allErrs
+}
+
+func validateV4InternalSubnet(n *types.Networking, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	v4IntNet := n.OVNKubernetesConfig.V4InternalSubnet
+	if v4IntNet == nil {
+		return allErrs
```

Comment:
There are no previous errors at this time.  Do we want to return nil?

Reply:
If v4InternalSubnet is not provided, it is not replaced with the default value at this time. If we do, the following validations would be performed on the default value of the subnet and provide early feedback on whether using the default would be OK. WDYT?

Reply:
> There are no previous errors at this time. Do we want to return nil?

+1 I think that is more readable. I incorporated the change.

> If v4InternalSubnet is not provided, it is not replaced with the default value at this time. If we do, the following validations would be performed on the default value of the subnet and provide early feedback on whether using the default would be OK. WDYT?

In general I avoid validating defaults, as they are regularly tested and it keeps things more maintainable. I think your suggestion is more about improving the validation of other fields (e.g. would user-supplied cluster network interfere with the default v4InternalSubnet), which is a good practice but out of scope for this feature.